src/java.base/share/classes/jdk/internal/foreign/abi/BindingSpecializer.java

```diff
@@ -65,69 +63,67 @@
     private static final int CLASSFILE_VERSION = ClassFileFormatVersion.latest().major();
 
-    private static final String OBJECT_DESC = Object.class.descriptorString();
-    private static final String OBJECT_INTRN = Type.getInternalName(Object.class);
-
-    private static final String VOID_DESC = methodType(void.class).descriptorString();
-
-    private static final String ARENA_DESC = Arena.class.descriptorString();
-    private static final String NEW_BOUNDED_ARENA_DESC = methodType(Arena.class, long.class).descriptorString();
-    private static final String NEW_EMPTY_ARENA_DESC = methodType(Arena.class).descriptorString();
-    private static final String SCOPE_DESC = methodType(MemorySegment.Scope.class).descriptorString();
-    private static final String SESSION_IMPL_DESC = methodType(MemorySessionImpl.class).descriptorString();
-    private static final String CLOSE_DESC = VOID_DESC;
-    private static final String UNBOX_SEGMENT_DESC = methodType(long.class, MemorySegment.class).descriptorString();
-    private static final String COPY_DESC = methodType(void.class, MemorySegment.class, long.class, MemorySegment.class, long.class, long.class).descriptorString();
-    private static final String LONG_TO_ADDRESS_NO_SCOPE_DESC = methodType(MemorySegment.class, long.class, long.class, long.class).descriptorString();
-    private static final String LONG_TO_ADDRESS_SCOPE_DESC = methodType(MemorySegment.class, long.class, long.class, long.class, MemorySessionImpl.class).descriptorString();
-    private static final String ALLOCATE_DESC = methodType(MemorySegment.class, long.class, long.class).descriptorString();
-    private static final String HANDLE_UNCAUGHT_EXCEPTION_DESC = methodType(void.class, Throwable.class).descriptorString();
-    private static final String METHOD_HANDLES_INTRN = Type.getInternalName(MethodHandles.class);
-    private static final String CLASS_DATA_DESC = methodType(Object.class, MethodHandles.Lookup.class, String.class, Class.class).descriptorString();
-    private static final String RELEASE0_DESC = VOID_DESC;
-    private static final String ACQUIRE0_DESC = VOID_DESC;
-    private static final String INTEGER_TO_UNSIGNED_LONG_DESC = MethodType.methodType(long.class, int.class).descriptorString();
-    private static final String SHORT_TO_UNSIGNED_LONG_DESC = MethodType.methodType(long.class, short.class).descriptorString();
-    private static final String BYTE_TO_UNSIGNED_LONG_DESC = MethodType.methodType(long.class, byte.class).descriptorString();
-
-    private static final Handle BSM_CLASS_DATA = new Handle(
-            H_INVOKESTATIC,
-            METHOD_HANDLES_INTRN,
-            "classData",
-            CLASS_DATA_DESC,
-            false);
-    private static final ConstantDynamic CLASS_DATA_CONDY = new ConstantDynamic(
-            ConstantDescs.DEFAULT_NAME,
-            OBJECT_DESC,
-            BSM_CLASS_DATA);
+    private static final ClassDesc CD_Arena = desc(Arena.class);
```

Comment:
I love that we're able to describe what we need to use at a higher-level of abstraction than just strings.